protostuff-api/src/main/java/io/protostuff/StringSerializer.java

```diff
@@ -318,7 +318,7 @@ public static LinkedBuffer writeDouble(final double value, final WriteSession se
     /**
      * Computes the size of the utf8 string beginning at the specified {@code index} with the specified {@code length}.
      */
-    public static int computeUTF8Size(final String str, final int index, final int len)
+    public static int computeUTF8Size(final CharSequence str, final int index, final int len)

```

Comment:
Would you rather I create new functions for StringBuilder? indirection can indeed impact perf, although I have not seen any regressions .. yet. Have been running for a day now and I see no difference in serialization latencies.